### DISCUSSION TITLE
Fuzzy matching precedence

### DISCUSSION THREAD

**Discussion by newadventure079** [cross-seed/cross-seed#171] [Q&A] [answered]
Does fuzzy matching comparison happen before or after the condition of "it has a different file tree"?

**Comment by mmgoodnow** [comment]
http://github.com/mmgoodnow/cross-seed/blob/b74cbcc9da69ae88af4efb1e73dbb522487f0973/src/decide.ts#L79-L103

**Comment by newadventure079** [comment]
If there's a torrent with an extra .nfo file, cross-seed will pass the return statement on line 85 (assuming the fuzzing matching is configured large enough), but it will still return FILE_TREE_MISMATCH on line 99, which is not the expected result.

**Reply by mmgoodnow** [reply]
PRs are welcome to rectify that

**Reply by newadventure079** [reply]
If I did submit a PR to rectify it, that would end up fixing #160, which is outside the scope of the project, correct?

**Reply by mmgoodnow** [reply, marked as answer]
oh yeah, sorry I wasn't paying enough attention. I thought you meant it wasn't working when the _searchee_ has an extra nfo file, but what you mean is that it won't work when the _candidate_ has an extra nfo file. I do think I'd like to keep it that way, because injecting those torrents into your client would eventually fail a hash check.

If you want to experiment with ways to improve match rate, there is one thing I'm not 100% sure about which is the following:

Browsing through your cache, you can see several torrents which reject with FILE_TREE_MISMATCH. Most of these are probably because of NFOs, but I have a suspicion that there might be some valid matches lurking around because of different torrents' piece sizes fuzzing file lengths. I'm not 100% sure however. 

I added two new commands to help with this sort of endeavor, `cross-seed diff` and `cross-seed tree` which you just pass torrent files to, and it'll spit out some information about them, which could help. 

Another way you could work on improving match rate is this searchee deduplication code [here](http://github.com/mmgoodnow/cross-seed/blob/d4502397280507dd88b0945ff48e7feab86d1d1f/src/preFilter.ts#L46-L54).
If you start out with a torrent containing an NFO (let's call this A), then it finds a cross-seed that _doesn't_ have an NFO (B) which has an alphabetically earlier info hash, then after that, you won't be able to match with other torrents containing the NFO (C) because B has now become the de-facto searchee for the whole group of torrents, and it doesn't contain the NFO, even though you do have the data because of A. 

A way around this would be to write a custom deduplication function using Array.prototype.reduce which prefers the torrent with more files in it
